 main.tf

```diff
@@ -0,0 +1,118 @@
+terraform {
+    required_providers {
+        azurerm = {
+            source = "hashicorp/azurerm"
+            version = ">= 2.49"
+            }
+    }
+        backend "azurerm" {
+        resource_group_name  = "CreditSuisse2_TasnimaMiah_ProjectExercise"
+        storage_account_name = "tfstatewqs"
+        container_name       = "tfstate"
+        key                  = "terraform.tfstate"
+    }
+}
+provider "azurerm" {
+    features {}
+}
+
+resource "azurerm_storage_account" "tfstate" {
+  name                     = "tfstatewqs"
+  resource_group_name      = data.azurerm_resource_group.main.name
+  location                 = var.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  allow_blob_public_access = true
+
+  tags = {
+    environment = "staging"
+  }
+}
+
+resource "azurerm_storage_container" "tfstate" {
+  name                  = "tfstate"
+  storage_account_name  = azurerm_storage_account.tfstate.name
+  container_access_type = "blob"
+}
+
+data "azurerm_resource_group" "main" {
+    name = "CreditSuisse2_TasnimaMiah_ProjectExercise"
+}
+resource "azurerm_app_service_plan" "main" {
+    name = "terraformed-asp"
+    location = data.azurerm_resource_group.main.location
+    resource_group_name = data.azurerm_resource_group.main.name
+    kind = "Linux"
+    reserved = true
+    sku {
+        tier = "Basic"
+        size = "B1"
+    }
+}
+resource "azurerm_app_service" "main" {
+    name = "devops-todo-app"
+    location = data.azurerm_resource_group.main.location
+    resource_group_name = data.azurerm_resource_group.main.name
+    app_service_plan_id = azurerm_app_service_plan.main.id
+    site_config {
+        app_command_line = ""
+        linux_fx_version = "DOCKER|tasnimamiah/my-test-image:latest"
+    }
+    app_settings = {
+        "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+        "MONGODB_CONNECTION_STRING" = "mongodb://${azurerm_cosmosdb_account.tasnimamiah.name}:${azurerm_cosmosdb_account.tasnimamiah.primary_key}@${azurerm_cosmosdb_account.tasnimamiah.name}.mongo.cosmos.azure.com:10255/DefaultDatabase?ssl=true&replicaSet=globaldb&retrywrites=false&maxIdleTimeMS=120000&appName=@tasnimamiah@"
+        "SECRET_KEY"="secret-key"
```

Comment:
Not important for this practice exercise, but for a real app the secret key should be treated like a secret, as the name implies. It's used to encrypt cookies and if it is leaked then anyone could pretend to be logged into your application as anyone else.